deploy/ods-pipeline/charts/tasks/templates/task-ods-deploy-helm.yaml

```diff
@@ -72,7 +72,16 @@ spec:
       type: string
       default: ./chart
     - name: release-name
-      description: The Helm release name. If empty, the release name is simply the name of the chart.
+      description: |
+        The Helm release name. 
+        If empty, the release name is simply the name of the chart. Furthermore in this case the
+        task sets `<subcomponent>.fullnameOverride` equal to the respective 
+        subcomponent to avoid resources being prefixed with the umbrella repository
+        component name (assuming your resources are named using the `chart.fullname`
+        helper).
+        In contrast a `release-name` is specified the `chart.fullname` helper typically
+        prefixes resources with the specfied `release-name` unless the chart's name contains
+        the `release-name`. 
```

Comment:
Maybe we can make this a new paragraph and say that it only applies when umbrella charts are used?

I am thinking something like the following:

```
The Helm release name. If empty, the release name is simply the name of the chart.

When this task is used in a repository which defines subcharts, and the parameter is not set,
then the task sets `<subcomponent>.fullnameOverride` equal to the respective subcomponent
to avoid resources being prefixed with the umbrella repository component name (assuming your
resources are named using the `chart.fullname` helper). However, if the parameter is specified,
`<subcomponent>.fullnameOverride` is not set. As a result, the `chart.fullname` helper prefixes
resources with the specified `release-name` unless the chart's name contains the `release-name`. 
```